sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -1536,21 +1536,26 @@ class Dataset[T] private[sql](
    * Returns a new Dataset by sampling a fraction of rows.
    *
    * @param withReplacement Sample with replacement or not.
-   * @param fraction Fraction of rows to generate.
+   * @param fraction Fraction of rows to generate and the range is 0.0 < `fraction` < 1.0.
    * @param seed Seed for sampling.
    *
    * @group typedrel
    * @since 1.6.0
    */
   def sample(withReplacement: Boolean, fraction: Double, seed: Long): Dataset[T] = withTypedPlan {
-    Sample(0.0, fraction, withReplacement, seed, logicalPlan)()
+    if (0.0 < fraction && fraction < 1.0) {
```

Comment:
I'm not sure about sampling methods though, is it natural that `sampling` methods have `fraction>1.0`?
Seems sampling naturally means it randomly picks the part of input data. Is this incorrect?